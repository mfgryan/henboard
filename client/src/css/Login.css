#Account {
    text-align: left;
}

#Account input {
    margin-top: 6px;
}

#Account h3 {
    margin-top: 50px;
}

.AccountField {
    margin-bottom: 10px
} 

#AccountSub {
    margin: 10px 12px 0 0;
}

.AccountButton {
    text-align: right;
} 
