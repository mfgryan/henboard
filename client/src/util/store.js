// redux dep
import { createStore, applyMiddleware } from "redux";
import { logger, validate, fn } from "../util/middleware.js"; // eslint-disable-line no-unused-vars
import rootReducer from "../reducers/reducers.js";

export default (function(){
    let store;
    return !store ? createStore(rootReducer, applyMiddleware(validate, logger, fn)) : store;
}());
