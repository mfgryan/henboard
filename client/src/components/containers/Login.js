// react dep
import React, { Component } from "react";
import { connect } from "react-redux";
import { login } from "../../actions/user.js";

// component dep
import { Col, Row, Button, Modal } from "react-bootstrap";

// style dep
import "../../css/Login.css";

class Login extends Component {
    constructor(props){
        super(props); 
        this.state = {
            email: "", 
            password: ""
        };
    }

    render() {
        let user = { email: this.state.email, password: this.state.password };
        return (
            <Modal.Dialog id="Account" draggable="false">
                <Modal.Body>
                    <Row>
                        <Col md={4} mdOffset={3}>
                            <h3>Login</h3>
                        </Col>
                    </Row>
                    <Row>
                        <Col md={2} mdOffset={3} >
                            <h4>Email</h4>
                        </Col>
                        <Col md={4}>
                            <input type="text" value={user.email} onChange={e => this.setState({ email: e.target.value}) }/>
                        </Col>
                    </Row>
                    <Row>
                        <Col md={2} mdOffset={3} >
                            <h4>Password</h4>
                        </Col>
                        <Col md={4}>
                            <input type="password" value={user.password} onChange={e => this.setState({password: e.target.value}) }/>
                        </Col>
                    </Row>
                    <Row>
                        <Col md={6} mdOffset={3} className="AccountButton">
                            <Button id="AccountSub" onClick={() => this.props.login(user)} >Submit</Button>
                        </Col>
                    </Row>
                    <br />
                </Modal.Body>
                <Modal.Footer>
                    <Row>
                        <Col md={1} mdOffset={10}>
                            <Button onClick={() => this.props.close()}>Close</Button>
                        </Col>
                    </Row>
                </Modal.Footer>
            </Modal.Dialog>
        );
    }
};

const mapStateToProps = state => {
    return {};
};

const mapDispatchToProps = dispatch => {
    return {
        login: (user) => {
            dispatch(login(user));
        }
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(Login);
