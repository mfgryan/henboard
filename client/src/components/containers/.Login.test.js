import React from "react";
import Login from "./Login";
import renderer from "react-test-renderer";

// redux dep
import { Provider } from "react-redux";
import store from "../../util/store.js";

// router dep
import { BrowserRouter as Router } from "react-router-dom";

it("Login renders correctly", () => {
    const tree = renderer
        .create(
            <Router>
                <Provider store={store}>
                    <Login close={() => {}}/>
                </Provider>
            </Router>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
