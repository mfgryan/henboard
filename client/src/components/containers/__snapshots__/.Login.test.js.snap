exports[`test Login renders correctly 1`] = `
<div
  className="modal"
  draggable="false"
  id="Account"
  role="dialog"
  style={
    Object {
      "display": "block",
    }
  }
  tabIndex="-1">
  <div
    className="modal-dialog">
    <div
      className="modal-content"
      role="document">
      <div
        className="modal-body">
        <div
          className="row">
          <div
            className="col-md-4 col-md-offset-3">
            <h3>
              Login
            </h3>
          </div>
        </div>
        <div
          className="row">
          <div
            className="col-md-2 col-md-offset-3">
            <h4>
              Email
            </h4>
          </div>
          <div
            className="col-md-4">
            <input
              onChange={[Function]}
              type="text"
              value="" />
          </div>
        </div>
        <div
          className="row">
          <div
            className="col-md-2 col-md-offset-3">
            <h4>
              Password
            </h4>
          </div>
          <div
            className="col-md-4">
            <input
              onChange={[Function]}
              type="password"
              value="" />
          </div>
        </div>
        <div
          className="row">
          <div
            className="AccountButton col-md-6 col-md-offset-3">
            <button
              className="btn btn-default"
              disabled={false}
              id="AccountSub"
              onClick={[Function]}
              type="button">
              Submit
            </button>
          </div>
        </div>
        <br />
      </div>
      <div
        className="modal-footer">
        <div
          className="row">
          <div
            className="col-md-1 col-md-offset-10">
            <button
              className="btn btn-default"
              disabled={false}
              onClick={[Function]}
              type="button">
              Close
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
